### PR TITLE
feat(publish): Use craft config from merge target if configured

### PR DESCRIPTION
**Pull request by Lms24**
This PR does 2 things:

1. Extract the merge target branch information added to the publish issue in https://github.com/getsentry/action-prepare-release/pull/35
2. Checkout the target repo on the merge target branch if we extracted `merge_target` from

In a follow up PR, I'll add an allowlist for repos and branches where we can take the config from and fall back if the merge target isn't included in the allowlist.

ref https://github.com/getsentry/publish/issues/3441

### CODE REVIEW

**Comment by Lms24** [comment]
@asottile-sentry can't request a review due to repo permissions I guess. Would you mind taking a look? thanks!

File: src/modules/details-from-context.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 async function detailsFromContext({ context }) {
   if (!context || !context.payload || !context.payload.issue) {
-    throw new Error('Issue context is not defined');
+    throw new Error("Issue context is not defined");
```
Comment (Lms24) [inline src/modules/details-from-context.js:3]: formatting change

File: src/modules/details-from-context.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
   }
 
-  const titleParser = /^publish: (?:getsentry\/)?(?<repo>[^/@]+)(?<path>\/[\w./-]+)?@(?<version>[\w.+-]+)$/;
+  const titleParser =
+    /^publish: (?:getsentry\/)?(?<repo>[^/@]+)(?<path>\/[\w./-]+)?@(?<version>[\w.+-]+)$/;
```
Comment (Lms24) [inline src/modules/details-from-context.js:7]: formatting change

File: src/modules/details-from-context.js [hunk 3]
```diff
@@ -14,10 +15,10 @@ async function detailsFromContext({ context }) {
   // - Cannot have multiple consecutive hyphens.
   // - Cannot begin or end with a hyphen.
   // - Maximum 39 characters.
-  const requesterParser = /^Requested by: @(?<requester>[a-zA-Z\d](?:[a-zA-Z\d]|-(?=[a-zA-Z\d])){0,38})$/m;
-  const { requester } = context.payload.issue.body.match(
-    requesterParser
-  ).groups;
+  const requesterParser =
```
Comment (Lms24) [inline src/modules/details-from-context.js:18]: formatting change

File: src/modules/details-from-context.js (hunk 2 above)
Comment (asottile-sentry) [reply src/modules/details-from-context.js:7]: please discard the unrelated formatting changes from your PR so I can review the actual change

**Comment by BYK** [comment]
>In a follow up PR, I'll add an allowlist for repos and branches where we can take the config from and fall back if the merge target isn't included in the allowlist.

This should come first for safety. I'd also say (as a next step to the allow list) that it would be better to check whether the branch name is covered by branch protection rules. I _think_ `secrets.GH_SENTRY_BOT_PAT` token would have access to read other repo settings. If not you can ask @HazAT or others in that chain to add that permission.

**Comment by Lms24** [comment]
Superseded by #3452

**Pull request closed without merging**